lambda_functions/healthcheck/healthcheck.py

```diff
@@ -0,0 +1,126 @@
+                    'Value': expected_string,
+                    },
+                    {
+                    'Name': 'ResponseStatus',
+                    'Value': str(status_code),
+                    },
+                    {
+                    'Name': 'FailureType',
+                    'Value': failure_type,
+                    },
+                ],
+                'Unit': 'Count',
+                'Value': 1,
+            },
+        ],
+        Namespace = 'SML-Healthcheck.'
+    )
+
+    print("Metric Data: ", metric_data)
+
+def check_web_url_health(url, env, expected_string):
+    """
+    check_web_url_health is a function that calls the url from the event
+    and checks if the status code is 200 and the expected string exists on that url.
+
+    :param url: Url of url to be checked
+    :type url: string
+    :param expected_string: String of text to be checked
+    :type expected_string: String
+    :param env: This is for us to distinguish the message between the dev, preprod and prod environments
+    :type env: string
+    :param failure_type: The failure type helps to distinguish between a response and content failure for metric collection.
+    :type failure_type: string
+    :raises HealthCheckException: this custom exception will trigger if the response does not match our expected values.
+    :raises HealthCheckException: error
+    """
+    response = requests.get(url, timeout=5)
+
+    # If the response code is not 200 or the response text does not 
+    # contain the expected string then we log an error and fail the lambda
+    if response.status_code != 200:
+        failure_type = "UnexpectedResponseCode"
+        failing_metric(url, env, expected_string, response.status_code, failure_type)
+        raise HealthCheckException(f"Status code expected to be 200 but is {response.status_code}")
+
+    elif expected_string not in response.text:
+        failure_type = "UnexpectedResponseContent"
+        failing_metric(url, env, expected_string, response.status_code, failure_type)
+        raise HealthCheckException(f"Status code is {response.status_code} but expected text \'{expected_string}\' is not found")
+    
+def lambda_handler(event, context):
+    """
+    lambda_handler takes variables from the event and calls the healthcheck function
+
+    :param event: Event is the message contents passes to the lambda e.g.
+                    {
+                        "expected_string" = "An open source library for statistical code approved by the ONS",
+                        "env" = "dev",
+                        "url" = "https://dev-sml.aws.onsdigital.uk/"
+                    }
+    :type event: string
+    :param context: General Lambda term (not used in this case but needed for general setup)
+    :type context: N/A
+    :raises Exception: will raise a error if there is an issue with the healthcheck.
+    :raises Exception: error
+    """    
+
+    print("Event: ", event)
```

Comment:
I don't think we want this - presumably this just fills the logs up with a print statement every minute?

Reply:
It does yes I can remove it :)